### PR TITLE
Core/Memmap: Give more detailed error messages if memory mapping fails.

### PR DESCRIPTION
**Pull request by AdmiralCurtiss**
Mostly I have a build to suggest over here to see if this could be a failure in InitFastmemArena(), which is silent on master: https://forums.dolphin-emu.org/Thread-wii-games-extremely-slow-after-revision-5-0-11476

### CODE REVIEW

File: Source/Core/Core/HW/Memmap.cpp [hunk 1]
```diff
@@ -379,7 +387,10 @@ void UpdateLogicalMemory(const PowerPC::BatTable& dbat_table)
           void* mapped_pointer = g_arena.CreateView(position, mapped_size, base);
           if (!mapped_pointer)
           {
-            PanicAlertFmt("MemoryMap_Setup: Failed finding a memory base.");
+            PanicAlertFmt(
+                "Memory::UpdateLogicalMemory(): Failed to create view for physical region at "
+                "0x{:08X} (size 0x{:08X}).",
+                intersection_start - mapping_address, mapped_size);
```
Comment (AdmiralCurtiss) [inline Source/Core/Core/HW/Memmap.cpp:393]: The physical address here might be wrong, I admit I don't fully understand what this code is doing...

File: Source/Core/Core/HW/Memmap.cpp [hunk 2]
```diff
@@ -299,7 +299,9 @@ void Init()
 
     if (!*region.out_pointer)
     {
-      PanicAlertFmt("MemoryMap_Setup: Failed finding a memory base.");
+      PanicAlertFmt(
```
Comment (leoetlino) [inline Source/Core/Core/HW/Memmap.cpp:302]: MsgHandler.h needs to be included.

**Review by leoetlino — CHANGES_REQUESTED** [review]

**Comment by AdmiralCurtiss** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/Core/HW/Memmap.cpp [hunk 3]
```diff
@@ -333,6 +339,9 @@ bool InitFastmemArena()
 
     if (base != view)
     {
+      PanicAlertFmt("Memory::InitFastmemArena(): Failed to map memory region at 0x{:08X} (size "
```
Comment (leoetlino) [inline Source/Core/Core/HW/Memmap.cpp:342]: Nitpicking, but can the `(size ` be moved to the next line to make this more pleasant? (I assume clang-format is what reformatted it like this.)

Comment (AdmiralCurtiss) [reply Source/Core/Core/HW/Memmap.cpp]: Yup that was clang-format. Should be nicer now.

**Review by leoetlino — APPROVED** [review]

**Pull request merged**